### DISCUSSION TITLE
I want to manually ADD ip address to block

### DISCUSSION THREAD

**Discussion by legalizecannabis** [celzero/rethink-app#568] [Ideas]
So, I have the older version (with the DRASTICALLY BETTER interface--the new one is absolutely, positively trash).  I should not have to scroll horizontally to see all options (dns types) when I could just click the arrow and get the drop down list, horrible design, more time-consuming and just plain downgraded usability.

So in the old version, I can see "blocked IPs" but cannot ADD any?  So I ran a DIG command on a telemetry IP from my tablet, and wanted to add it, because the firewall (for some unknown reason) does not at all catch it.  I have all "apps" blocked, except browser and some music / podcast program.  Still, this domain gets through.  This is an Amazon tablet, so I don't have this issue on other android devices, just this one.

Also, on my tablet, I am absolutely FORCED to use 8.8.8.8 by the "DNS" app / program built into the system, even though I have set two custom in settings, it adds a THIRD dns entry against my ability to change and prevents ALL internet use.  This was usually not an issue, and I don't think I've updated the android OS (I think I have all that blocked), but lately it's been consistent, and so I have to use a NextDNS to bypass and have a cloud-based firewall to get past this.

I would like to see an option to use the "old" UI because it's 1,000,000 times better, easier, and I don't have to select both wi-fi and mobile data--at least make an option to just do both for all connections please, tapping both is annoying.

Also lastly, and most importantly, back to the NextDNS stuff, if I choose just DNS proxy, it just doesn't work, 8.8.8.8 still gets requested and so.

**Comment by ignoramous** [comment]
>  I have all "apps" blocked, except browser and some music / podcast program. Still, this domain gets through. 

On Android, the OS itself sends DNS queries on behalf of the apps (this is par for the course for most OSes), and so, even if you block an app, there's no way to block the DNS queries per-app because, from Rethink's point-of-view, it is the OS that's requesting those queries.

That said, We do have a trick coming up in `v054` (a month or two away) that might just work in also unearthing the app actually sending DNS queries #270 

> So, I have the older version (with the DRASTICALLY BETTER interface--the new one is absolutely, positively trash). ... I would like to see an option to use the "old" UI because it's 1,000,000 times better, easier, and I don't have to select both wi-fi and mobile data--at least make an option to just do both for all connections please, tapping both is annoying.

Sorry but we had to rid of the older UI as it was too complicated. Based on your feedback, however; we've tried to further improve the current UI where bulk firewall actions are now possible. Can you take a look at the [latest version `v053n`](https://rethinkdns.com/download) and let us know if you like it?

> Also lastly, and most importantly, back to the NextDNS stuff, if I choose just DNS proxy, it just doesn't work, 8.8.8.8 still gets requested and so.

The app may request `8.8.8.8` but if you've "Prevent DNS Leaks" enabled (check DNS *Configure* UI), those requests to `8.8.8.8` and `8.8.4.4` are re-routed to the DNS upstream of your choosing. You can see this by tapping on the "Network Log" entries corresponding to `8.8.8.8` or `8.8.4.4` and it should have a green-coloured chip in the top right-hand corner of the bottomsheet with text "DNS Proxied".
